code/modules/research/server.dm

```diff
@@ -4,44 +4,70 @@
 	icon = 'icons/obj/machines/research.dmi'
 	icon_state = "RD-server-on"
 	var/datum/techweb/stored_research
-	var/heat_health = 100
 	//Code for point mining here.
-	var/working = TRUE			//temperature should break it.
+	var/overheated = FALSE			//temperature should break it.
+	var/working = TRUE
 	var/research_disabled = FALSE
 	var/server_id = 0
-	var/base_mining_income = 2
-	var/current_temp = 0
-	var/heat_gen = 100
-	var/heating_power = 40000
-	var/delay = 5
-	var/temp_tolerance_low = 0
+	// some notes on this number
+	// as of 4/29/2020, the techweb was set that fed a constant of 52.3 no matter how many servers there were
+	// A coeffecent of sqrt(100/<servercount>) is set up on a per some older code.  Since there are normaly 2 servers this comes out to
+	// sqrt(100/2) = 7.07, then 52.3 /  7.07 = 7.40.  Since we have two servers per map, these are added together
+	// 7.40./2 = 3.70 (note, all these values are rounded).  This is howw this number was found.
+	var/base_mining_income = 3.70
+
+	// Heating is wierd.  Since  the servers are stored in a room that sucks air in one vent, into a pipe network, to a
+	// T1 freezer, then out another vent at standard presure, the rooms temps could vary as wieldy as 100K.  The T1 freezer
+	// has 10000 heat power at the start, so each of the servers produce that but only heat a quarter of the turf
+	// This allows the servers to rapidly heat up in under 5 min to the shut off point and make it annoying to cool back
+	// down, giving time for RD to fire the guy who shut off the cooler
+
+	var/heating_power = 10000		// Changed the value from 40000.  Just enough for a T1 freezer to keep up with 2 of them
+	var/heating_effecency = 0.25
+	var/temp_tolerance_low = T0C
 	var/temp_tolerance_high = T20C
+	var/temp_tolerance_damage = T0C + 200		// Most CPUS get up to 200C they start breaking.  TODO: Start doing damage to the server?
 	var/temp_penalty_coefficient = 0.5	//1 = -1 points per degree above high tolerance. 0.5 = -0.5 points per degree above high tolerance.
+	var/datum/component/thermo/thermo
```

Comment:
You should not save references to components like this, its generally bad practice and you should be listening to events instead. In this case you would need a OnRefreshParts signal on /obj/machinery/proc/RefreshParts()